selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/SeleniumWebDriverHelper.java

```diff
@@ -12,69 +12,461 @@
+   * Waits until provided {@link WebElement} be visible
+   *
+   * @param webElement
+   * @return found element
+   */
+  public WebElement waitElementIsVisible(WebElement webElement) {
+    return waitElementIsVisible(webElement, DEFAULT_TIMEOUT);
+  }
+
+  /**
+   * Waits until {@link WebElement} with provided locator {@link By} be attached to DOM it does not
+   * mean that element is visible
+   *
+   * @param elementLocator
+   * @param timeout waiting time in seconds
+   * @return found element
+   */
+  public WebElement waitElementIsPresentInDom(By elementLocator, int timeout) {
+    return webDriverWaitFactory.get(timeout).until(presenceOfElementLocated(elementLocator));
+  }
+
+  /**
+   * Waits until {@link WebElement} with provided locator {@link By} be attached to DOM it does not
+   * mean that element is visible
+   *
+   * @param elementLocator
+   * @return found element
+   */
+  public WebElement waitElementIsPresentInDom(By elementLocator) {
+    return webDriverWaitFactory
+        .get(DEFAULT_TIMEOUT)
+        .until(presenceOfElementLocated(elementLocator));
+  }
+
+  /**
+   * Type text in the {@link WebElement} with provided locator {@link By}
+   *
+   * @param elementLocator
+   * @param text
+   */
+  public void sendKeysToElement(By elementLocator, String text) {
+    waitElementIsVisible(elementLocator).sendKeys(text);
+  }
+
+  /**
+   * Type text in the provided {@link WebElement}
+   *
+   * @param webElement
+   * @param text
+   */
+  public void sendKeysToElement(WebElement webElement, String text) {
+    waitElementIsVisible(webElement).sendKeys(text);
+  }
+
+  /**
+   * Waits visibility of {@link WebElement} with provided locator {@link By} and get text
+   *
+   * @param fieldLocator
+   * @return element text by {@link WebElement#getAttribute(String)}
+   */
   public String getFieldValue(By fieldLocator) {
-    return waitAndGetElement(fieldLocator).getAttribute("value");
+    return waitElementIsVisible(fieldLocator).getAttribute("value");
+  }
+
+  /**
+   * Waits visibility of provided {@link WebElement} and get text
+   *
+   * @param webElement
+   * @return element text by {@link WebElement#getAttribute(String)}
+   */
+  public String getFieldValue(WebElement webElement) {
+    return waitElementIsVisible(webElement).getAttribute("value");
+  }
+
+  /**
+   * Waits visibility of {@link WebElement} with provided locator {@link By} and get text
+   *
+   * @param fieldLocator
+   * @return element text by {@link WebElement#getText()}
+   */
+  public String getFieldText(By fieldLocator) {
+    return waitElementIsVisible(fieldLocator).getText();
   }
 
-  public WebElement waitAndGetElement(By locator) {
-    return loadPageWait.until(visibilityOfElementLocated(locator));
+  /**
+   * Waits visibility of provided {@link WebElement} and get text
+   *
+   * @param webElement
+   * @return element text by {@link WebElement#getText()}
+   */
+  public String getFieldText(WebElement webElement) {
```

Comment:
`waitElementText` describes the propose of method much better, IMHO

Reply:
Sorry but I am not agree, this method waits for visibility of webelement and get text from it. It does not wait for a particular text.

Reply:
`waitElementIsVisibleAndGetText`